packages/graphql/src/translate/translate-top-level-union-read.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Union } from "../classes/Union";
+import Node from "../classes/Node";
+import { Context } from "../types";
+import translateRead from "./translate-read";
+
+function translateToplevelUnionRead({ union, context }: { union: Union; context: Context }): [string, any] {
+    const cypher: string[] = [];
+    const returnEntries: string[] = [];
+    let cypherParams = {};
+
+    Object.entries(context.resolveTree.fieldsByTypeName).forEach((entry, i) => {
+        if (entry[0] === union.name) {
+            return;
+        }
+
+        const toNode = context.nodes.find((n) => n.name === entry[0]) as Node;
+        const whereInput = context.resolveTree.args?.where?.[toNode.name] as Record<string, unknown>;
```

Comment:
to make sure the types match correctly, we should add `undefined` to these types
```suggestion
        const whereInput = context.resolveTree.args?.where?.[toNode.name] as Record<string, unknown> | undefined;
```